### PR TITLE
fix(KFLUXBUGS-1182): update infra deployment use target git repo from data

### PR DESCRIPTION
**Pull request by happybhati**
- Modified `update-infra-deployments` task to dynamically fetch
  `targetGHRepo`, `githubAppID`, and `githubAppInstallationID` from the dataJsonPath JSON file.
  (defaults to redhat-appstudio/infra-deployments for targetGHRepo,
  and provided defaults for githubAppID and githubAppInstallationID)
- Added `defaultTargetGHRepo`, `defaultGithubAppID`, and `defaultGithubAppInstallationID` parameters
  to the update-infra-deployments task to specify the default values
  for the GitHub repository and app configurations.

### CODE REVIEW

File: tasks/update-infra-deployments/update-infra-deployments.yaml
Comment (mmalina) [inline tasks/update-infra-deployments/update-infra-deployments.yaml]: Nit: `."targetGHRepo"` can be just `.targetGHRepo`. The apostrophes are only needed if there are special characters like `-` in the field name.

File: tasks/update-infra-deployments/update-infra-deployments.yaml
Comment (johnbieren) [inline tasks/update-infra-deployments/update-infra-deployments.yaml]: I think we should go to 1.0.0 here - removing a parameter is a breaking change

File: tasks/update-infra-deployments/update-infra-deployments.yaml
Comment (johnbieren) [inline tasks/update-infra-deployments/update-infra-deployments.yaml:1]: What do you think about leaving the default as a parameter so we only have the default defined once? The code would stay exactly as you have it now, you'd just replace the spots where redhat-appstudio/infra-deployments is written with the params.paramName piece

**Review by johnbieren — COMMENTED** [review]
Can you add a test?

File: tasks/update-infra-deployments/update-infra-deployments.yaml
Comment (happybhati) [reply tasks/update-infra-deployments/update-infra-deployments.yaml]: ACK, thanks!

File: tasks/update-infra-deployments/update-infra-deployments.yaml
Comment (happybhati) [reply tasks/update-infra-deployments/update-infra-deployments.yaml:1]: Got it, thanks!

**Comment by happybhati** [comment]
> Can you add a test?

I initially added a test, although it was broken.

Martin commented: 
> In its current form, I don’t think it’s feasible to make a test for this
> 
> the issue is that it has a step with a Python script included inline. Ideally, the Python script would be saved in the utils image, and then you could mock it.

We may need to refactor the task in a separate story. Or @johnbieren  if you have any idea?

**Comment by johnbieren** [comment]
> > Can you add a test?
> 
> I initially added a test, although it was broken.
> 
> Martin commented:
> 
> > In its current form, I don’t think it’s feasible to make a test for this
> > the issue is that it has a step with a Python script included inline. Ideally, the Python script would be saved in the utils image, and then you could mock it.
> 
> We may need to refactor the task in a separate story. Or @johnbieren if you have any idea?

That's a great point. You can forget my comment about tests then

File: tasks/update-infra-deployments/README.md
Comment (johnbieren) [inline tasks/update-infra-deployments/README.md]: Maybe one more bullet point about the new parameter

**Review by johnbieren — COMMENTED** [review]
One comment in the README, but otherwise lgtm
Can you test this locally before we merge? I am pretty sure you will be fine because the `$(params.paramName)` notation substitutes the value beforehand so quoting acts normal, but I am not entirely sure and would feel better if we were

**Review by johnbieren — APPROVED** [review]
I saw you moved this to draft for a bit with other changes and now it is back to open. Does that mean you tested it and it works? The code LGTM assuming that is the case

**Comment by happybhati** [comment]
> I saw you moved this to draft for a bit with other changes and now it is back to open. Does that mean you tested it and it works? The code LGTM assuming that is the case

@johnbieren Yes, the PR was tested with Leandro's help, and we were also able to generate a pull request by providing a repo and config in the RPA data field. Thanks for the review.

Thanks @theflockers for helping with the test.

**Pull request merged**
